src/core/mesh/qgsmeshlayer.cpp

```diff
@@ -277,8 +277,10 @@ void QgsMeshLayer::assignDefaultStyleToDatasetGroup( int groupIndex )
 
 QgsMapLayerRenderer *QgsMeshLayer::createMapRenderer( QgsRenderContext &rendererContext )
 {
+  bool nativeMeshHasChanged = dataProvider()->meshHasChanged();
```

Comment:
`nativeMeshHasChanged` should be signal and we should have slot `onNativeMeshChanged` which will effectively reload mesh and all datasets.